parsl/dataflow/dflow.py

```diff
@@ -587,15 +621,18 @@ def checkpoint(self):
                     pickle.dump(t, f)
                     count += 1
                     self.tasks[task_id]['checkpoint'] = True
-                    logger.debug("Task {}: checkpointed".format(task_id))
+                    logger.debug("Task {} checkpointed".format(task_id))
+
+        self.checkpointed_tasks += count
 
-        end = time.time()
         if count == 0:
-            logger.warn(
-                'No tasks checkpointed, please ensure caching is enabled')
+            if self.checkpointed_tasks == 0:
+                logger.warn("No tasks checkpointed, please ensure caching is enabled")
```

Comment:
Can we just check if caching is enabled automatically and return an error if someone tries to instantiate the DFK this way?

Reply:
This is harder. We can test for appCaching being enabled at the top_level, but we won't be able to find out if specific apps have caching enabled. This is why that warning is added.

Reply:
Sounds good, thanks for explaining.